sncosmo/models.py

```diff
@@ -997,6 +998,317 @@ def colorlaw(self, wave=None):
+    modeldir : str, optional
+        Directory path containing model component files. Default is `None`,
+        which means that no directory is prepended to filenames when
+        determining their path.
+    m0file, m1file, clfile : str or fileobj, optional
+        Filenames of various model components. Defaults are:
+
+        * m0file = 'salt2_template_0.dat' (2-d grid)
+        * m1file = 'salt2_template_1.dat' (2-d grid)
+        * clfile = 'salt2_color_correction.dat'
+
+    errscalefile, lcrv00file, lcrv11file, lcrv01file, cdfile : str or fileobj
+        (optional) Filenames of various model components for
+        model covariance in synthetic photometry. See
+        ``bandflux_rcov`` for details.  Defaults are:
+
+        * errscalefile = 'salt2_lc_dispersion_scaling.dat' (2-d grid)
+        * lcrv00file = 'salt2_lc_relative_variance_0.dat' (2-d grid)
+        * lcrv11file = 'salt2_lc_relative_variance_1.dat' (2-d grid)
+        * lcrv01file = 'salt2_lc_relative_covariance_01.dat' (2-d grid)
+        * cdfile = 'salt2_color_dispersion.dat' (1-d grid)
+
+    Notes
+    -----
+    The "2-d grid" files have the format ``<phase> <wavelength>
+    <value>`` on each line.
+
+    The phase and wavelength values of the various components don't
+    necessarily need to match. (In the most recent salt2 model data,
+    they do not all match.) The phase and wavelength values of the
+    first model component (in ``m0file``) are taken as the "native"
+    sampling of the model, even though these values might require
+    interpolation of the other model components.
+
+    """
+
+    _param_names = ['x0', 'x1', 'c']
+    param_names_latex = ['x_0', 'x_1', 'c']
+    _SCALE_FACTOR = 1e-12
+
+    def __init__(self, modeldir=None,
+                 m0file='salt3_template_0.dat',
+                 m1file='salt3_template_1.dat',
+                 clfile='salt3_color_correction.dat',
+                 cdfile='salt3_color_dispersion.dat',
+                 errscalefile='salt3_lc_dispersion_scaling.dat',
+                 lcrv00file='salt3_lc_variance_0.dat',
+                 lcrv11file='salt3_lc_variance_1.dat',
+                 lcrv01file='salt3_lc_covariance_01.dat',
+                 name=None, version=None):
+        self.name = name
+        self.version = version
+        self._model = {}
+        self._parameters = np.array([1., 0., 0.])
+
+        names_or_objs = {'M0': m0file, 'M1': m1file,
+                         'LCRV00': lcrv00file, 'LCRV11': lcrv11file,
+                         'LCRV01': lcrv01file, 'errscale': errscalefile,
+                         'cdfile': cdfile, 'clfile': clfile}
+
+        # Make filenames into full paths.
+        if modeldir is not None:
+            for k in names_or_objs:
+                v = names_or_objs[k]
+                if (v is not None and isinstance(v, str)):
+                    names_or_objs[k] = os.path.join(modeldir, v)
+
+        # model components are interpolated to 2nd order
```

Comment:
I think of "2nd order" as quadratic, but this is using cubic interpolation. Is there a a wording mismatch?

Reply:
this is from the original SALT2 - I don't understand it but I'm hesitant to change it, given that it could indicate something about how the model is generated.  I removed this from the new code because I realized the only thing I really need to change is how the errors are generated, so I just made SALT3Source an instance of SALT2Source and overwrote the error method.